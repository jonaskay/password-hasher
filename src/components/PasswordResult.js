import React, { Component } from 'react';
import './PasswordResult.css';

class PasswordResult extends Component {
  constructor(props) {
    super(props);
    this.selectTextInput = this.selectTextInput.bind(this);
  }

  selectTextInput() {
    this.textInput.select();
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevProps.result !== this.props.result) {
      this.selectTextInput();
    }
  }

  render() {
    return (
      <div className="PasswordResult">
        <input
          type="text"
          value={this.props.result}
          ref={input => {
            this.textInput = input;
          }}
          readOnly
        />
      </div>
    );
  }
}

export default PasswordResult;
