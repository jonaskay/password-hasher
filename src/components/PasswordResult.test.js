import React from 'react';
import { shallow } from 'enzyme';
import PasswordResult from './PasswordResult';

describe('PasswordResult', () => {
  it('renders correctly', () => {
    expect(shallow(<PasswordResult result="foobar" />)).toMatchSnapshot();
  });
});
