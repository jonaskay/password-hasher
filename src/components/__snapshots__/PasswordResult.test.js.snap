// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordResult renders correctly 1`] = `
<div
  className="PasswordResult"
>
  <input
    readOnly={true}
    type="text"
    value="foobar"
  />
</div>
`;
